File: libgomp/testsuite/libgomp.oacc-c-c++-common/private-vars-loop-gang-1.c
#include <assert.h>

/* Test of gang-private variables declared on loop directive.  */

int
main (int argc, char* argv[])
{
  int x = 5, i, arr[32];

  for (i = 0; i < 32; i++)
    arr[i] = i;

  #pragma acc parallel copy(arr) num_gangs(32) num_workers(8) vector_length(32)
  {
    #pragma acc loop gang private(x)
    for (i = 0; i < 32; i++)
      {
	x = i * 2;
	arr[i] += x;
      }
  }

  for (i = 0; i < 32; i++)
    assert (arr[i] == i * 3);

  return 0;
}

File: libgomp/testsuite/libgomp.oacc-fortran/private-vars-loop-gang-1.f90
! { dg-additional-options "-w" }

! Test of gang-private variables declared on loop directive.

program main
  integer :: x, i, arr(32)

  do i = 1, 32
     arr(i) = i
  end do

  !$acc parallel copy(arr) num_gangs(32) num_workers(8) vector_length(32)
  !$acc loop gang private(x)
  do i = 1, 32
     x = i * 2;
     arr(i) = arr(i) + x
  end do
  !$acc end parallel

  do i = 1, 32
     if (arr(i) .ne. i * 3) call abort
  end do
end program main
